rust/candid/src/de.rs

```diff
@@ -74,7 +74,7 @@ impl<'de> IDLDeserialize<'de> {
                 TypeInner::Opt(_) | TypeInner::Reserved | TypeInner::Null
             ) {
                 self.de.expect_type = expected_type;
-                self.de.wire_type = TypeInner::Reserved.into();
+                self.de.wire_type = TypeInner::Null.into();
```

Comment:
What does this actually do?

Reply:
This sets a "virtual" wire type for values that are actually missing on the wire: in this case, we're decoding an element in a top-level tuple that does not correspond to a value on the wire, e.g., decoding the last element in a tuple of type `(nat, nat, null)` with the tuple `(5:nat, 6:nat)` actually on the wire. In this case, the type of such an extra element in the tuple must be `null`, `opt`, or `reserved`, but only `null` is a subtype of all these types and thus the only legal choice in general (`reserved` used to work before because it did coerce into `null`, too, which does PR is fixing).